src/_pytest/fixtures.py

```diff
@@ -268,16 +272,23 @@ def reorder_items_atscope(
                     i for i in scoped_items_by_argkey[slicing_argkey] if i in items
                 ]
                 for i in reversed(matching_items):
-                    fix_cache_order(i, argkeys_cache, items_by_argkey)
                     items_deque.appendleft(i)
+                    # Fix items_by_argkey order.
+                    for other_scope in HIGH_SCOPES:
+                        other_scoped_items_by_argkey = items_by_argkey[other_scope]
+                        for argkey in argkeys_by_item[other_scope].get(i, ()):
+                            other_scoped_items_by_argkey[argkey][i] = None
+                            other_scoped_items_by_argkey[argkey].move_to_end(
```

Comment:
@bluetech I've hit some corner case on PyPy where `move_to_end()` seems to be raising a `KeyError` and crashing the item collection entirely...

Reply:
Oh, nevermind. This is a regression of v8.2.2 and this PR only hit release in v8.3.0.